src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -1594,6 +1618,15 @@ int LinearScan::BuildPutArgStk(GenTreePutArgStk* putArgStk)
 
     srcCount = BuildOperandUses(src);
     buildInternalRegisterUses();
+
+#ifdef TARGET_X86
+    // There are only 4 byteable registers on x86. If we require a byteable internal register,
+    // we must have less than 4 sources.
+    // If we have 4 or more sources, and require a byteable internal register, we need to reserve
+    // one explicitly (see BuildBlockStore()).
+    assert(srcCount < 4);
```

Comment:
This and the other assert below are added because these are the only other semi-complex cases where we have byteable restrictions. I've convinced myself that we will never have more than 3 uses in these cases, so a simple assert is useful and sufficient (we don't need to check that we *actually* have a byteable register source).